poly-synth/src/poly-synth.js

```diff
@@ -72,13 +90,101 @@ class PolySynth {
       filterDecay: 1,
       filterSustain: this.minSustain,
       filterRelease: this.minRelease,
-      filterDetuneAmount: 1
+      filterDetuneAmount: 1,
     };
 
+    // Each voice is connected to |this.voiceOutput_|.
+    this.voiceOutput_ = new GainNode(this.context_);
+
+    // The output of |this.voiceOutput_| is processed by three effects
+    // bitcrusher, feedback delay, and a convolver.
+    let feedbackInputNode = new GainNode(this.context_);
+    let feedbackOutputNode = this.createFeedbackGraph_(feedbackInputNode);
+
+    let bitcrusherInputNode = new GainNode(this.context_);
+    let bitcrusherOutputNode = this.createBitcrusherGraph_(bitcrusherInputNode);
+
+    let convolverInputNode = new GainNode(this.context_);
+    let convolverOutputNode = this.createConvolverGraph_(convolverInputNode);
+
     // The client is responsible for connecting |this.output| to a destination.
     this.output = new GainNode(this.context_);
+
+    // The feedback node occurs prior to the other effects so that the output
+    // of the feedback can be modulated by these components.
+    this.voiceOutput_.connect(feedbackInputNode);
+    feedbackOutputNode.connect(bitcrusherInputNode);
+    bitcrusherOutputNode.connect(convolverInputNode);
+    convolverOutputNode.connect(this.output);
+  }
+
+  createFeedbackGraph_(inputNode) {
+    // The sum of the gain of the wet and dry signal is always 1, and the output
+    // always routes through both nodes.
+    this.feedbackWetSignal_ =
+        new GainNode(this.context_, {gain: this.feedbackWetness});
+    this.feedbackDrySignal_ =
+        new GainNode(this.context_, {gain: 1 - this.feedbackWetness});
+    this.feedbackDelayNode_ =
+        new DelayNode(this.context_, {delayTime: this.feedbackDelayTime});
+    this.feedbackGainNode_ =
+        new GainNode(this.context_, {gain: this.feedbackGain});
+    let feedbackEffectOutputNode = new GainNode(this.context_, {gain: 1});
+
+    inputNode.connect(this.feedbackDrySignal_)
+        .connect(feedbackEffectOutputNode);
+
+    // The output of the delay node routes through the wet path as well as a
+    // gain node which echoes the delay node's output back to it at a gain
+    // specified by the user.
+    inputNode.connect(this.feedbackDelayNode_)
+        .connect(this.feedbackWetSignal_)
+        .connect(feedbackEffectOutputNode);
+    this.feedbackDelayNode_.connect(this.feedbackGainNode_)
+        .connect(this.feedbackDelayNode_);
+
+    return feedbackEffectOutputNode;
   }
-  
+
+  createBitcrusherGraph_(inputNode) {
+    this.bitcrusher = new Bitcrusher(
+        this.context_,
+        {bitDepth: this.maxBitDepth, reduction: this.minReduction});
+
+    // Only one of |this.activeBitcrusherRoute_| and
+    // |this.bypassBitcrusherRoute_| has a non-zero gain
```

Comment:
This isn't really true, right? You do linear-ramp these so at some point both are non-zero.